File: src/main/scala/test/person.scala
package test

class Person(
              navn: String,
              etternavn: String,
              alder: Int) {
  val interesser = List("ting", "og", "tang")
  val cv = "%s %s -- %s\n Liker: %s".format(navn, etternavn, alder, interesser.mkString(";"))

  def fulltNavn = {
    "%s %s".format(navn, etternavn)
  }

  def mineInteresser = {
    var interesser = ""
    for (interesse <- interesser) {
      interesser += interesse + ", "
    }

    interesser
  }
}


class App {
  //Scala liste opperator ::
  val etTog = new Vogn :: new Vogn :: new Vogn :: Nil

  class Vogn {
    def +(vogn: Vogn) = List(this, vogn)

    def ::(vogn: Vogn) = List(this, vogn)
  }

  // Vår egen syntaks
  val tog = new Vogn + new Vogn
  val tog2 = new Vogn :: new Vogn :: new Vogn
  println(tog2)


  val liste = List("en", "to", "tre")

  def prat2Rop(tekst: String) = "%s!".format(tekst)
  liste.map(prat2Rop(_))

  var ropeList = List[String]()
  for(i <- liste){
    ropeList = "%s!".format(i) :: ropeList
  }


object Logger{
  def error(msg:String) = println ("ÆÆÆ! : " + msg)
}

trait Loggable{
  def error(msg: String) = Logger.error(msg)
}

class ServiceA extends Loggable{  }
class ServiceB extends Loggable{  }



  val list = List(1,2,3)
  list(0) == 1 //samme som list.apply(0)

  val to = 1 + 1 // sammen som 1.+(1)

}

File: src/main/java/test/Person.java

import java.util.List;
import java.util.ArrayList;

public class Person{
    private String navn;
    private String etternavn;
    private Integer alder;

    public Person(String fornavn, String etternavn, Integer alder){
        this.navn = fornavn;
        this.etternavn = etternavn;
        this.alder = alder;
    }

    public String fulltNavn(){
        return navn + " " + etternavn;
    }

    public List<String> interesse(String starterMed){
        List<String> interesser = new ArrayList<String>();
        interesser.add("Gym");
        interesser.add("Sykkel");
        interesser.add("Scala");
        interesser.add("Vask");
        interesser.add("Mat");

        List<String> valgte = new ArrayList<String>();
        for(String interesse : interesser){
            if(interesse.startsWith(starterMed)){
                valgte.add(interesse);
            }
        }

        return valgte;
    }

    public String getNavn(){ return navn; }
    public void setNavn(String navn){ this.navn = navn; }

    public String getEtternavn(){ return etternavn; }
    public void setEtternavn(String etternavn){ this.etternavn = etternavn; }

    public Integer getAlder(){ return alder; }
    public void setAlder(Integer alder){ this.alder = alder; }
}


